content/guides/printing/pprint.adoc

```diff
@@ -0,0 +1,161 @@
+#'user/foo
+user=> (binding [*print-length* 6] (pprint (foo 10)))
+((1 2 3 4 5 6 ...)
+ (1 2 3 4 5 6 ...)
+ (1 2 3 4 5 6 ...)
+ (1 2 3 4 5 6 ...)
+ (1 2 3 4 5 6)
+ (1 2 3 4 5)
+ ...)
+nil
+user=>
+----
+
+=== Custom Dispatch Functions
+Using custom dispatch, you can easily create your own formatted output for structured data.
+Examples included with the pretty printer show how to use custom dispatch to translate simple Clojure structures into nicely formatted JSON and XML.
+
+=== Basic Concepts of Pretty Printing
+In order to create custom dispatch functions, you need to understand the fundamentals of pretty printing.
+The clojure pretty printer is based on the XP pretty printer algorithm (used in many Lisps including Common Lisp) which supports sophisticated
+decision-making about line breaking and indentation with reasonable performance even for very large structures. The
+XP algorithm is documented in the paper, http://dspace.mit.edu/handle/1721.1/6504[XP]. A Common Lisp Pretty Printing System.
+
+The Clojure implementation of XP is similar in spirit to the Common Lisp implementation, but the details of the interface are somewhat different.
+The result is that writing custom dispatch in Clojure is more "Clojure-y."
+
+There are three key concepts to understand when creating custom pretty printing functions: `logical blocks`, `conditional newlines`, and `indentation`.
+
+A logical block marks a set of output that should be thought about as a single unit by the pretty printer. Logical blocks can contain other logical
+blocks (that is, they nest). As a simple example, when printing list structure, every sublist will typically be a logical block.
+
+Conditional newlines tell the pretty printer where it can insert line breaks and how to make the decisions about when to do it. There are four types
+of conditional newline:
+
+* Linear newlines tell the pretty printer to insert a newline in a place whenever the enclosing logical block won't fit on a single line. Linear newlines are an all-or-nothing proposition; if the logical block doesn't fit on a single line, all the linear newlines are emitted as actual newlines.
+
+* Fill newlines tell the pretty printer that it should fit as many chunks of the logical block as possible on this line and then emit a newline.
+Mandatory newlines tell the pretty printer to emit a newline regardless of where it is in the output line.
+
+* Miser newlines tell the pretty printer to emit a newline if the output column is in the miser region (as defined by the pretty printer variable *pprint-miser-width*). This allows you to define special behavior as the output gets heavily nested near the right margin.
+
+* Indentation commands allow you to specify how wrapped lines should be indented. Indentation can be relative to either the start column of the current logical block or the current column position of the output.
+
+(This section is still incomplete...)
+
+=== Current limitations and future plans
```

Comment:
I would remove this or at couch it as "future areas for improvement" rather than "not done".